v2/datastream-to-spanner/src/main/java/com/google/cloud/teleport/v2/templates/spanner/ProcessInformationSchema.java

```diff
@@ -98,6 +98,8 @@ static class ProcessInformationSchemaFn extends DoFn<Void, Ddl> {
     // Timeout for Cloud Spanner schema update.
     private static final int SCHEMA_UPDATE_WAIT_MIN = 5;
 
+    private boolean useSeparateShadowTableDb = false;
```

Comment:
nit: would it not be easier to just pass around a null shadowTableConfig for shadow table and use that null check instead of this additional flag ?

Reply:
[nit and not a blocker in anyway] +1 for having a single variable represent both. We could consider [java.util.Optional](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Optional.html) as well.
Dataflow would need @nullable marked at many places if we are passing around something that could be null.

Reply:
Sure will make these imporvements in the IT pr.